sdk/storage/azure_storage_blob/tests/page_blob_client.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use azure_core::http::{RequestContent, StatusCode};
+use azure_core_test::{recorded, TestContext};
+use azure_storage_blob::models::{
+    BlobClientDownloadResultHeaders, BlobClientGetPropertiesResultHeaders, BlobType,
+    PageBlobClientCreateOptions,
+};
+use azure_storage_blob_test::{get_blob_name, get_container_client};
+use std::error::Error;
+
+#[recorded::test]
+async fn test_create_page_blob(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    // Recording Setup
+    let recording = ctx.recording();
+    let container_client = get_container_client(recording, true).await?;
+    let blob_client = container_client.blob_client(get_blob_name(recording));
+    let page_blob_client = blob_client.page_blob_client();
+
+    // Regular Create Scenario
+    page_blob_client.create(1024, None).await?;
+    // Assert
+    let blob_properties = blob_client.get_properties(None).await?;
+    let blob_type = blob_properties.blob_type()?;
+    let content_length = blob_properties.content_length()?;
+    assert_eq!(1024, content_length.unwrap());
+    assert_eq!(BlobType::PageBlob, blob_type.unwrap());
+
+    // Create If Not Exists Scenario
+    let create_options = PageBlobClientCreateOptions {
+        if_none_match: Some("*".to_string()),
+        ..Default::default()
```

Comment:
Wondering if there is any cool ways we could make a helper for this i.e. like somehow have a `.if_not_exists()` that you can call on the options bag -- another pattern we haven't yet discussed (how to go about modifying customer's options bag)